src/runtime/hexagon_host.cpp

```diff
@@ -656,6 +662,77 @@ WEAK int halide_hexagon_power_hvx_on(void *user_context) {
     return 0;
 }
 
+WEAK int halide_hexagon_power_hvx_on_mode(void *user_context, halide_hvx_power_mode_t mode) {
+    int result = init_hexagon_runtime(user_context);
+    if (result != 0) return result;
+
+    debug(user_context) << "halide_hexagon_power_hvx_on_mode\n";
+    if (!remote_power_hvx_on_mode) {
+        // The function is not available in this version of the
+        // runtime, this runtime always powers HVX on.
```

Comment:
I think it is strictly necessary to call it, otherwise the reference count will be incorrect. Consider this sequence of calls on an old runtime that doesn't have the new APIs:

    halide_hexagon_power_hvx_on_mode(...);
    ...
    halide_hexagon_power_hvx_off(...);

On an old runtime that doesn't support the new APIs, the power on call will not increase the reference count, but the power off call will.